File: example/language/hello.c
#include <stdio.h>
#include <stdlib.h>

// C语言示例 - Hello World
int main(int argc, char *argv[]) {
    printf("Hello, C World!\n");

    // 变量声明
    int number = 42;
    float pi = 3.14159;
    char *message = "This is a test";

    // 条件语句
    if (number > 40) {
        printf("Number is greater than 40\n");
    } else {
        printf("Number is not greater than 40\n");
    }

    // 循环
    for (int i = 0; i < 5; i++) {
        printf("Iteration: %d\n", i);
    }

    // 函数调用
    return EXIT_SUCCESS;
}

// 函数定义
void example_function(void) {
    // 这是一个示例函数
    return;
}

File: example/language/hello.nim
import strformat

type
  Person = object
    name: string
    age: int

proc greet(person: Person): string =
  fmt"Hello, {person.name}! You are {person.age} years old."

proc main() =
  let person = Person(name: "World", age: 2024)
  echo greet(person)

  # Simple loop example
  for i in 1..5:
    echo fmt"Count: {i}"

  # Conditional example
  let number = 42
  if number > 40:
    echo "Number is greater than 40"
  elif number > 30:
    echo "Number is greater than 30"
  else:
    echo "Number is small"

when isMainModule:
  main()
